adam-core/src/main/scala/org/bdgenomics/adam/models/SingleReadBucket.scala

```diff
@@ -49,6 +53,114 @@ case class SingleReadBucket(primaryMapped: Iterable[AlignmentRecord] = Seq.empty
   def allReads = {
     primaryMapped ++ secondaryMapped ++ unmapped
   }
+
+  def toFragment: Fragment = {
+    // take union of all reads, as we will need this for building and
+    // want to pay the cost exactly once
+    val unionReads = allReads
+
+    // do we have any chimeric reads?
+    val sequences: Seq[Sequence] = if (secondaryMapped.exists(_.getSupplementaryAlignment)) {
```

Comment:
I would really like someone else to go through the chimeric read reconstruction code with a fine toothed comb. The logic here wasn't fun to work through, and has a few edge cases. If there are any sections that aren't clear to people who aren't me, I would like to add additional comments to clarify them.